core/Network/TLS/Parameters.hs

```diff
@@ -194,14 +195,23 @@ data Supported = Supported
       --   enough until 2030.  Both curves are fast because their
       --   backends are written in C.
     , supportedGroups              :: [Group]
+    , supportedEarlyData           :: EarlyData
     } deriving (Show,Eq)
 
+data EarlyData = NoEarlyData
+               | AcceptEarlyData Word32
+               | SendEarlyData ByteString
+               deriving (Eq, Show)
+
```

Comment:
- `NoEarlyData` is deleted. `AcceptEarlyData 0` is used instead.
- `AcceptEarlyData Word32` is changed to `AcceptEarlyData Int`
- Document is written.

Reply:
OK but if type is different for client and server, maybe this should be moved out of Supported to ClientParams / ServerParams directly.

Reply:
`AcceptEarlyData` is used in `recvData` which takes only `Context`.
So, if we take your approach, we need to add a new field to `Context` and copy the value from `ServerParams` to `Context`.

It it worth doing?

Reply:
Yes that looks useful. Because it would allow to track how much 0RTT data has been received and how much left is allowed. Maybe inside the `EarlyDataAllowed` constructor.

Reply:
Using `EarlyDataAllowed` is a really nice idea!
The patch to implement this idea is pushed.